cmd/coreos-assembler.go

```diff
@@ -0,0 +1,170 @@
+package main
+
+import (
+	_ "embed"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"sort"
+	"strings"
+	"syscall"
+)
+
+// commands we'd expect to use in the local dev path
+var buildCommands = []string{"init", "fetch", "build", "run", "prune", "clean", "list"}
+var advancedBuildCommands = []string{"buildfetch", "buildupload", "oc-adm-release", "push-container", "upload-oscontainer"}
+var buildextendCommands = []string{"aliyun", "aws", "azure", "digitalocean", "exoscale", "gcp", "ibmcloud", "kubevirt", "live", "metal", "metal4k", "nutanix", "openstack", "qemu", "secex", "virtualbox", "vmware", "vultr"}
+var utilityCommands = []string{"aws-replicate", "compress", "generate-hashlist", "koji-upload", "kola", "remote-build-container", "remote-prune", "sign", "tag"}
+var otherCommands = []string{"shell", "meta"}
+
+func init() {
+	// Note buildCommands is intentionally listed in frequency order
+	sort.Strings(advancedBuildCommands)
+	sort.Strings(buildextendCommands)
+	sort.Strings(utilityCommands)
+	sort.Strings(otherCommands)
+}
+
+func wrapCommandErr(err error) error {
+	if err == nil {
+		return nil
+	}
+	if exiterr, ok := err.(*exec.ExitError); ok {
+		return fmt.Errorf("%w\n%s", err, exiterr.Stderr)
+	}
+	return err
+}
+
+func printCommands(title string, cmds []string) {
+	fmt.Printf("%s:\n", title)
+	for _, cmd := range cmds {
+		fmt.Printf("  %s\n", cmd)
+	}
+}
+
+func printUsage() {
+	fmt.Println("Usage: coreos-assembler CMD ...")
+	printCommands("Build commands", buildCommands)
+	printCommands("Advanced build commands", advancedBuildCommands)
+	printCommands("Platform builds", buildextendCommands)
+	printCommands("Utility commands", utilityCommands)
+	printCommands("Other commands", otherCommands)
+}
+
+func run(argv []string) error {
+	if err := initializeGlobalState(argv); err != nil {
+		return fmt.Errorf("failed to initialize global state: %w", err)
+	}
+
+	var cmd string
+	if len(argv) > 0 {
+		cmd = argv[0]
+		argv = argv[1:]
+	}
+
+	if cmd == "" {
+		printUsage()
+		os.Exit(1)
+	}
+
+	target := fmt.Sprintf("/usr/lib/coreos-assembler/cmd-%s", cmd)
+	_, err := os.Stat(target)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return fmt.Errorf("unknown command: %s", cmd)
+		}
+		return fmt.Errorf("failed to stat %s: %w", target, err)
+	}
+
+	c := exec.Command(target, argv...)
+	c.Stdin = os.Stdin
+	c.Stdout = os.Stdout
+	c.Stderr = os.Stderr
+	if err := c.Run(); err != nil {
+		return fmt.Errorf("failed to execute cmd-%s: %w", cmd, err)
+	}
+	return nil
+}
+
+func initializeGlobalState(argv []string) error {
+
+	// Set PYTHONUNBUFFERED=1 so that we get unbuffered output. We should
+	// be able to do this on the shebang lines but env doesn't support args
+	// right now. In Fedora we should be able to use the `env -S` option.
+	os.Setenv("export PYTHONUNBUFFERED", "1")
```

Comment:
```suggestion
	os.Setenv("PYTHONUNBUFFERED", "1")
```

Otherwise I'm getting:

```
Error: error parsing host environment variables: name "export PYTHONUNBUFFERED" has white spaces, poorly formatted name
```